cpp/oneapi/dal/algo/decision_forest/backend/gpu/test/feature_type_helper.cpp

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include "oneapi/dal/test/engine/dataframe.hpp"
+#include "oneapi/dal/algo/decision_forest/backend/gpu/feature_type_helper.hpp"
+
+namespace oneapi::dal::decision_forest::test {
+
+namespace de = dal::detail;
+namespace be = dal::decision_forest::backend;
+namespace te = dal::test::engine;
+
+template <typename TestType>
+class indexed_feature_test : public te::float_algo_fixture<std::tuple_element_t<0, TestType>> {
+public:
+    using Float = std::tuple_element_t<0, TestType>;
+    using Bin = std::tuple_element_t<1, TestType>;
+    using Index = std::tuple_element_t<2, TestType>;
+
+public:
+    bool is_cpu() {
+        return this->get_policy().is_cpu();
+    }
+
+    auto get_dataframe_base() {
+        constexpr std::int64_t row_count = 6;
+        constexpr std::int64_t column_count = 3;
+
+        static const float data_arr[] = { -2.f, -9.f, 0.f, -1.f, -1.f, 0.f, -1.f, -2.f, 0.f,
+                                          +1.f, +1.f, 1.f, +1.f, +2.f, 1.f, +2.f, +1.f, 1.f };
+
+        static const Index bin_count_arr[] = { 4, 5, 2 };
+
+        static const Float bin_border_arr[] = { -2.f, -1.f, 1.f, 2.f, -9.f, -2.f,
+                                                -1.f, 1.f,  2.f, 0.f, 1.f };
+
+        te::dataframe data{ array<float>::wrap(data_arr, row_count * column_count),
+                            row_count,
+                            column_count };
+
+        return std::tuple(data, bin_count_arr, bin_border_arr);
+    }
+
+    auto base_checks(const te::dataframe& data,
+                     const te::table_id& data_table_id,
+                     const Index* bin_count_arr_ref,
+                     const Float* bin_border_arr_ref) {
+        auto& q = this->get_queue();
+        const auto tbl = data.get_table(data_table_id, range(0, -1));
+
+        be::indexed_features<Float, Bin, Index> iftrs(q, 5, 256);
+
+        iftrs(tbl).wait_and_throw();
```

Comment:
I suggest to have this at the level of TEST. There is a good pattern for unit tests:
```c++
TEST() {
   // Block 1: Preparation (constant definitions, data generation, etc...)
   // blank line
   // Block 2: Function under test (create object, call some method you want to test)
   // blank line
   // Block 3: Checks
}
```

To summarize taking in account comment regarding SRP:
```c++
TEST_M(indexed_feature_test, "basic checks") {
    using indexed_features_t = typename indexed_feature_test<TestType>::indexed_features_t;
    SKIP_IF(this->is_cpu());

    const std::int64_t min_bin_size = 5; // Add GENERATE here?
    const std::int64_t max_bin_count = 256; // Add GENERATE here?
    const auto [data, bin_count_ref, bin_border_ref] = this->get_dataframe_base();

    indexed_features_t ifeatures{ this->get_queue(), min_bin_size , max_bin_count };
    ifeatures(data).wait_and_throw();

    SECTION("check bin count") {
        this->check_bin_count(ifeatures, data, this->get_homogen_table_id(), bin_count_ref);
    }

    SECTION("check bin borders") {
        this->check_bin_borders(ifeatures, data, this->get_homogen_table_id(), bin_border_ref);
    }
}
```